controllers/databasecluster_controller.go

```diff
@@ -328,14 +329,23 @@ func (r *DatabaseClusterReconciler) genPSMDBBackupSpec(
 		switch backupStorage.Spec.Type {
 		case everestv1alpha1.BackupStorageTypeS3:
 			storages[backup.Spec.BackupStorageName] = psmdbv1.BackupStorageSpec{
-				Type: psmdbv1.BackupStorageType(backupStorage.Spec.Type),
+				Type: psmdbv1.BackupStorageS3,
 				S3: psmdbv1.BackupStorageS3Spec{
 					Bucket:            backupStorage.Spec.Bucket,
 					CredentialsSecret: backupStorage.Spec.CredentialsSecretName,
 					Region:            backupStorage.Spec.Region,
 					EndpointURL:       backupStorage.Spec.EndpointURL,
 				},
 			}
+		case everestv1alpha1.BackupStorageTypeAzure:
+			storages[backup.Spec.BackupStorageName] = psmdbv1.BackupStorageSpec{
```

Comment:
what about gcs support for psmdb?

Reply:
Per our discussion, psmdb does not have GCS support yet.